guillotina/request.py

```diff
@@ -239,18 +238,18 @@ def __init__(self, receive):
             raise asyncio.IncompleteReadError(data, n)
         return data
 
-    async def read(self, n: int = -1) -> bytes:
+    async def read(self, n: int = -1) -> Union[bytes, bytearray]:
         data = self._buffer
         while (n == -1 or len(data) < n) and not self._eof:
             chunk = await self.receive()
             data.extend(chunk)
 
         if n == -1:
             self._buffer = bytearray()
-            return bytes(data)
+            return data
         else:
             self._buffer = data[n:]
-            return bytes(data[:n])
+            return data[:n]
```

Comment:
I assume there is no need to convert `bytearray` to `bytes` Both types have the same interface. Also, I guess most of the time this is transformed to json isn't needed the conversion.